pkg/apis/sessions/session_state.go

```diff
@@ -102,20 +102,10 @@ func DecodeSessionState(v string, c encryption.Cipher) (*SessionState, error) {
 			PreferredUsername: ss.PreferredUsername,
 		}
 	} else {
-		// Backward compatibility with using unencrypted Email
-		if ss.Email != "" {
-			decryptedEmail, errEmail := stringDecrypt(ss.Email, c)
-			if errEmail == nil {
-				ss.Email = decryptedEmail
-			}
-		}
-		// Backward compatibility with using unencrypted User
-		if ss.User != "" {
-			decryptedUser, errUser := stringDecrypt(ss.User, c)
-			if errUser == nil {
-				ss.User = decryptedUser
-			}
-		}
+		// Backward compatibility with using unencrypted Email or User
+		// Decryption errors will leave original string
+		c.DecryptInto(&ss.Email)
```

Comment:
@JoelSpeed So I think this (and all previous implementations) are problematic -- but I'm thinking its not worth fixing because the error is rare and the need to support this will go away once we stop allowing SHA1 signed session cookies:

Since AES-CFB isn't an authenticated encryption, there's likely a subset of values that will successfully Base64 decode & then AES decrypt into a valid string (where it just assumed the first bytes are the IV vs part of the data) -> and this won't raise an error (which would then leave the User/Email as-is assuming they were legacy unencrypted). And instead they will be mangled into garbage.

Reply:
I'd prefer if you could write these as `_ := c.DecryptInto` to make it more obvious that this can error, but we are deliberately ignoring it (in addition to the comment preferably)